CRS-JEDI-G2-Flipkart-POS/src/com/flipkart/DAO/ProfessorDAOImpl.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * 
+ */
+package com.flipkart.DAO;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import com.flipkart.bean.CourseCatalog;
+import com.flipkart.bean.Professor;
+import com.flipkart.bean.Student;
+/**
+ * @author jagru
+ *
+ */
+public class ProfessorDAOImpl implements ProfessorDAOInterface {
+	
+	private static volatile ProfessorDAOImpl instance = null;
+	DAOConnectionInterface instanceDAO = new DAOConnectionImpl();
+	
+	/**
+	 * Method to make ProfessorDAOImpl Singleton
+	 * @return
+	 */
+	public static ProfessorDAOImpl getInstance()
+	{
+		if(instance == null)
+		{
+			synchronized(ProfessorDAOImpl.class){
+				instance = new ProfessorDAOImpl();
+			}
+		}
+		return instance;
+	}
+	
+	public ArrayList<CourseCatalog> viewOptedCourses(String professorId) throws SQLException {
+
+		String SelectOptedCoursesProf = "select * from 'courseCatalog' where profId = ?";
```

Comment:
@Jagruthi2001 Please make this name as SELECT_OPTED_COURSE 
final String SELECT_OPTED_COURSE = "select * from 'courseCatalog' where profId = ?";